src/query/service/src/sql/planner/optimizer/heuristic/prune_unused_columns.rs

```diff
@@ -60,9 +60,8 @@ impl UnusedColumnPruner {
                         "prewhere columns should be a subset of scan columns"
                     );
                     // `used` is the columns which prewhere scan needs to output for its upper operator.
-                    if used.is_empty() {
-                        let prewhere_columns =
-                            pw.prewhere_columns.iter().copied().collect::<Vec<_>>();
+                    let prewhere_columns = pw.prewhere_columns.iter().copied().collect::<Vec<_>>();
+                    if used.is_empty() && !prewhere_columns.is_empty() {
```

Comment:
I think `!prehwere_column.is_empty()` should always be `true`.

Reply:
For cases in the issue https://github.com/datafuselabs/databend/issues/7949, it's empty

Reply:
Oh, I know. It;s because the predicate in the where clause is a constant, so prehwere optimizer didn't collect it.

Reply:
Yes, for the case, we can fallback to `find_smallest_column_by_table_index`